src/modules/common/db/core/DBProvider.tsx

```diff
@@ -8,7 +10,24 @@ interface DBProviderProps {
 const Context = createContext({})
 
 export const DBProvider: FC<DBProviderProps> = ({ children }) => {
-  const db = useDBConnection()
+  const { dataSource, isInitialized, error } = useDBConnection()
 
-  return <Context.Provider value={db}>{children}</Context.Provider>
+  logger.debug('debug')
+  logger.info('info')
+  logger.warn('warn')
+  logger.error('error')
+
+  if (error) {
+    logger.error('error', error)
+  }
+
```

Comment:
you need to use useEffect to subscribe on changes with this variabel

Reply:
we register the error synchronously in the body of the component, so useEffect may be unnecessary and give nothing.

Reply:
```
 if (error) {
    logger.error('error', error)
  }
  ```
  This part can also be removed, as we just duplicate the error output which is logged in the hook.